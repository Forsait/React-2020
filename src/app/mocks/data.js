export const film = {
  budget: 0,
  genres: ['Action', 'Adventure', 'Drama', 'Science Fiction', 'Thriller'],
  id: 351286,
  overview: 'A volcanic eruption threatens',
  poster_path: 'https://image.tmdb.org/t/p/w500/ln6d5Okr6VK5vfQVobJTiYxeD0l.jpg',
  release_date: '2018-06-01',
  revenue: 0,
  runtime: null,
  tagline: 'Life finds a way',
  title: 'Jurassic World: Fallen Kingdom',
  vote_average: 0,
  vote_count: 28,
};

export const moviesData = {
  data: new Array(6).fill(film),
  limit: 0,
  offset: 0,
  total: 0,
};

export const radioOptions = {
  title: 'Radio title',
  list: [
    {
      key: 'key 1',
      title: 'title 1',
    },
    {
      key: 'key 2',
      title: 'title 2',
    },
  ],
};
