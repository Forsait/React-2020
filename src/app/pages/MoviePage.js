import React from 'react';
import { connect } from 'react-redux';

import MovieInfo from '../containers/movie/MovieInfo';
import MovieList from '../containers/movie/MovieList';
import Footer from '../components/common/Footer';
import { moviesListGoodSelector } from '../selectors';

class MoviePage extends React.Component {

  constructor(props){
    super(props);
  }

  render() {
    return <>
      <MovieInfo />
      <MovieList movieArr={this.props.moviesList}/>
      <Footer />
    </>
  }

}


function mapStateToProps(state) {
  return { moviesList: moviesListGoodSelector(state.movieInfo) };
}

export default connect(mapStateToProps)(MoviePage);