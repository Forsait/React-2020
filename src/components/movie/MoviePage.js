import React from 'react';

import styles from './MoviePage.module.scss';
import MovieList from './MovieList';
import Footer from '../common/Footer';
import Brand from '../common/Brand';

import searchIcon from '../../assets/icon-search.svg';

export default class MoviePage extends React.Component {

  constructor(props){
    super(props);

    this.state = {
      movie: {},
      movieList: [],
      releaseYear: '',
    };
  }

  componentDidMount() {
    this.loadMovieInfo();
  }

  loadMovieInfo() {
    const url = `/api/movies/` + 353081;
    fetch(url)
    .then(res => res.json())
    .then(res => {
      this.setState({
        movie: res,
        releaseYear: this.getReleaseYear(res.release_date),
      });
      console.log('movie', res);
      this.loadMoviesByGenre(res.genres[0])
    })
  }

  loadMoviesByGenre(genre) {
    const url = `/api/movies?offset=0&limit=6&sortBy=release_date&search=${genre}&searchBy=genres&sortOrder=desc`;

    fetch(url)
      .then(res => res.json())
      .then((res) => {
        console.log(res)
        this.setState({
          movieList: res.data,
        });
      })
  }

  getReleaseYear(dateStr) {
    if (!dateStr) return '';
    return dateStr.slice(0, 4);
  }

  render() {

    let genreView = '';
    if (this.state.movie.genres) {
      genreView = <div>Film by <span className={styles.som}>{this.state.movie.genres[0]}</span> genre</div>
    }

    return <>
      <div className={styles.container}>
        <div className={styles.sub_header}>
          <Brand />
          <a href="#" className={styles.search_link}>
            <img src={searchIcon} />
          </a>
        </div>

        <div className={styles.movie}>
          <img src={this.state.movie.poster_path} className={styles.poster}/>
          <div className={styles.info}>
            <div className={styles.title_box}>
              <h1 className={styles.title}>{this.state.movie.title}</h1>
              <span className={styles.rating}>{this.state.movie.vote_average}</span>
            </div>
            <div className={styles.subtitle}>Oscar winning movie</div>
            <div className={styles.maus}>
              <div className={styles.t4}>
                <span className={styles.t4_1}>{this.state.releaseYear}</span>
                <span className={styles.t4_2}>year</span>
              </div>
              <div className={styles.t4}>
                <span className={styles.t4_1}>{this.state.movie.runtime || '-'}</span>
                <span className={styles.t4_2}>min</span>
              </div>
            </div>
            <div className={styles.description}>{this.state.movie.overview}</div>
          </div>
        </div>
      </div>
      <div className={`${styles.hot} westside`}>
        {genreView}
      </div>
      <MovieList movieArr={this.state.movieList}/>
      <Footer />
    </>
  }

}