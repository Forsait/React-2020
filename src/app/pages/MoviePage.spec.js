import React from 'react';
import { shallow } from 'enzyme';
import { act } from "react-dom/test-utils";

import MoviePage from './MoviePage';
import { film } from '../mocks/data';

describe('Home component', () => {

  const serviceResponses = {
    '/api/movies/': film,
    '/api/movies?': new Array(6).fill(film),
  };

  it('should be render correctly', async () => {
    const data = new Array(6).fill(film);
    global.fetch = jest.fn().mockImplementation((url) => Promise.resolve({
        json: () => {
          for (let key in serviceResponses) {
            if(url.indexOf(key) !== -1) {
              return Promise.resolve(serviceResponses[key]);
            }
          }
        }
      })
    );

    await act(async () => {
      const com = shallow(<MoviePage />);
      expect(com).toMatchSnapshot()
    });

    global.fetch.mockClear();
    delete global.fetch;
  })

});