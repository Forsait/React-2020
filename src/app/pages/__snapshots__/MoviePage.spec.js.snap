// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should be render correctly 1`] = `
<Fragment>
  <MovieInfo
    movie={Object {}}
  />
  <MovieList
    movieArr={Array []}
  />
  <Footer />
</Fragment>
`;
